libraries/conf_template.rb

```diff
@@ -0,0 +1,85 @@
+# coding: UTF-8
+#
+# Cookbook Name:: cerner_splunk
+# File Name:: conf_template.rb
+
+module CernerSplunk
+  # Module for generating template data
+  module ConfTemplate
+    # Evaluate a Proc (or a Proc returning Procs)
+    def self.collapse_proc(initial, depth: 10, arguments: {}, label: nil)
```

Comment:
Depth doesn't actually ever exceed 1 currently... I wrote this method first, with only vague ideas of how I was going to using it. The thought was there could be situations where we may want to enable procs that return procs which return values and so on, but doing so without a depth option opens the possibility to an infinite loop. `10` was indeed an arbitrary choice, but I have crazy partial thoughts on scripted generation of configuration.